### PR TITLE
Add order status graph

### PR DESCRIPTION
**Pull request by krzysztofwolski**
Add documentation on possible status changes in Order object.

Related issue: #870

### CODE REVIEW

**Comment by timuric** [comment]
@krzysztofwolski looking good. Is it consumable as it is right now?

**Comment by szczecha** [comment]
Expired orders can be deleted after some time, more details: https://github.com/saleor/saleor/issues/12544 But I think it can be added in a separate PR

**Review by aniav — COMMENTED** [review]
In general it looks good! My comments are mostly about the mermaid structure and moving payment things to a separate document similar to the one about orders ✨

File: sidebars.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ module.exports = {
             "developer/checkout/order-to-checkout",
           ],
         },
+        "developer/order-status",
```
[the remark this answers is not in the document]
Comment (peelar) [reply sidebars.js]: done

File: docs/developer/order-status.mdx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+title: Order status
+---
+
+## Introduction
+
+A typical order lifecycle begins with a customer finalizing the checkout process and ends with delivering the products. Between those two points, the order can go through several states. This article describes the possible states of an order and the transitions between them.
+
+## Settings
+
+Besides the order status, several settings that can affect the behavior of the order. You can find them in the [`channel.orderSettings`](api-reference/miscellaneous/objects/order-settings.mdx) object.
```
[the remark this answers is not in the document]
Comment (peelar) [reply docs/developer/order-status.mdx]: done

File: docs/developer/order-status.mdx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    UNCONFIRMED --> EXPIRED: Expire time overdue
+    UNFULFILLED --> if_state_fulfillments: Fulfillment created
+    PARTIALLY_FULFILLED --> FULFILLED: All products fulfilled
+    if_state_fulfillments --> FULFILLED: All products fulfilled
+    if_state_fulfillments --> PARTIALLY_FULFILLED: Subset of products fulfilled
+    PARTIALLY_FULFILLED --> if_state_returns: Products returned
+
+
+    UNFULFILLED --> if_state_returns: Products returned
+    FULFILLED --> if_state_returns: Products returned
```
[the remark this answers is not in the document]
Comment (peelar) [reply docs/developer/order-status.mdx:43]: I think choice state in Mermaid can't be labeled unfortunately https://github.com/mermaid-js/mermaid/issues/3170

File: docs/developer/order-status.mdx [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+```mermaid
+stateDiagram-v2
+    direction LR
+
+    [*] --> UNCONFIRMED
+
+    UNCONFIRMED --> UNFULFILLED: Order confirmed
+    UNCONFIRMED --> EXPIRED: Expire time has passed
+    UNCONFIRMED --> CANCELED: Cancel the order
```
[the remark this answers is not in the document]
Comment (peelar) [reply docs/developer/order-status.mdx:86]: done

File: versioned_docs/version-3.x/developer/order-status.mdx [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+    UNFULFILLED --> CANCELED: Cancel the order
+
+    UNFULFILLED --> if_state_returns: Return items
+    if_state_returns --> RETURNED: All products returned
+    if_state_returns --> PARTIALLY_RETURNED: Not all products returned
+
+```
+
+- Item stocks are allocated.
```
[the remark this answers is not in the document]
Comment (peelar) [reply versioned_docs/version-3.x/developer/order-status.mdx:122]: @krzysztofwolski can you please have a look? 🙏

File: versioned_docs/version-3.x/developer/order-status.mdx [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+
+- Stock allocations are released.
+- Result of [`orderCancel`](api-reference/orders/mutations/order-cancel.mdx).
+- Possible previous states: UNCONFIRMED, UNFULFILLED.
+
+#### Available actions
+
+This state cannot be changed.
+
+## Payment status
```
[the remark this answers is not in the document]
Comment (peelar) [reply versioned_docs/version-3.x/developer/order-status.mdx]: I moved payment status to [a separate PR](https://github.com/saleor/saleor-docs/pull/1018) to not block this one

**Comment by krzysztofwolski** [comment]
"Refactored" order state graph to use flowchart instead of State graph, WDYT? @aniav @peelar 

```mermaid
flowchart TD
    CreatedDraft[Draft order created] --> DRAFT(DRAFT)
    CheckoutCompleted[Checkout Completed]

    DRAFT -->|Confirm order| UNFULFILLED(UNFULFILLED)

    CheckoutCompleted --> isAutomated{Is automated \norder confirmation turned on?}
    isAutomated -->|Yes| UNFULFILLED
    isAutomated -->|No| UNCONFIRMED(UNCONFIRMED)


    UNCONFIRMED -->|Expire time overdue| EXPIRED(EXPIRED)
    UNCONFIRMED -->|Confirm order| UNFULFILLED

    UNFULFILLED -->|Create fulfillment| isNotFulfilled{Are there unfulfilled\n order lines?}
    isNotFulfilled -->|Yes| PARTIALLY_FULFILLED(PARTIALLY_FULFILLED)
    isNotFulfilled -->|No| FULFILLED(FULFILLED)
    UNFULFILLED -->|Cancel order| CANCELLED(CANCELLED)

    PARTIALLY_FULFILLED -->|Create fulfillment| isNotFulfilled
    PARTIALLY_FULFILLED -->|Return products| isNotReturned{Are there unreturned\n order lines?}
    
    FULFILLED -->|Return products| isNotReturned
    isNotReturned -->|Yes| PARTIALLY_RETURNED(PARTIALLY_RETURNED)
    isNotReturned -->|No| RETURNED(RETURNED)
  
    PARTIALLY_RETURNED -->|Return products| isNotReturned

```

File: versioned_docs/version-3.x/developer/order-status.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply versioned_docs/version-3.x/developer/order-status.mdx]: There was mistake with draft order - no allocations at this step.
Also added mention of the track Inventory flag

File: docs/developer/order-status.mdx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+    state if_state_fulfillments <<choice>>
+    state if_state_returns <<choice>>
+    state if_state_automatic_confirmation <<choice>>
+
+    [*] --> DRAFT: Draft order created
+
+    DRAFT --> UNFULFILLED: Confirmed
+    [*] --> if_state_automatic_confirmation: Checkout completed
+    if_state_automatic_confirmation --> UNFULFILLED: With automatic confirmation
+    if_state_automatic_confirmation --> UNCONFIRMED: Without automatic confirmation
```
[the remark this answers is not in the document]
Comment (krzysztofwolski) [reply docs/developer/order-status.mdx]: Modifed type of graph, should be more clear now

**Pull request merged**